pkg/topology/full/full.go

```diff
@@ -77,6 +77,14 @@ func (d *driver) AddPeer(ctx context.Context, addr swarm.Address) error {
 	if !isConnected(addr, connectedPeers) {
 		peerAddr, err := d.p2pService.Connect(ctx, ma)
 		if err != nil {
+			d.mtx.Lock()
+			delete(d.receivedPeers, addr.ByteString())
+			d.mtx.Unlock()
+			var e *p2p.ConnectionBackoffError
+			if errors.Is(err, e) {
+				d.backoff(e.TryAfter())
```

Comment:
not sure if I missed anything, but, how is it that you're first declaring a nil pointer to `*p2p.ConnectionBackoffError`, then checking if `err` is of the same type (ok) but then calling `d.backoff(e.TryAfter())`. isn't this a null pointer exception?

Reply:
It should be type assertion, my bad 🙈